### PR TITLE
make the propertiesmanager store properties correctly

### PR DESCRIPTION
**Pull request by horrible-little-slime**
At present, the PropertiesManager doesn't store properties properly. I would like it to do so.

### CODE REVIEW

File: src/property.ts [hunk 1]
```diff
@@ -215,27 +215,22 @@ export class PropertiesManager {
...
 
   set(propertiesToSet: Properties): void {
-    this.properties = {
-      ...Object.keys(propertiesToSet).map((propertyName) => [
-        propertyName,
-        get(propertyName),
-      ]),
-      ...this.properties,
-    };
     Object.entries(propertiesToSet).forEach(([propertyName, propertyValue]) => {
```
Comment (phulin) [inline src/property.ts:225]: Can you use for...of instead of forEach? Better style.

File: src/property.ts [hunk 2]
```diff
@@ -215,27 +215,22 @@ export class PropertiesManager {
...
   setChoices(choicesToSet: { [choice: number]: number | string }): void {
-    this.set(
-      Object.fromEntries(
-        Object.entries(choicesToSet).map(([choiceNumber, choiceValue]) => [
-          `choiceAdventure${choiceNumber}` as NumericOrStringProperty,
-          choiceValue,
-        ])
-      )
-    );
+    Object.entries(choicesToSet).forEach(([choiceNumber, choiceValue]) => {
```
Comment (phulin) [inline src/property.ts:227]: Why not use this.set? DRY

File: src/property.ts [hunk 3]
```diff
@@ -215,27 +215,22 @@ export class PropertiesManager {
...
   set(propertiesToSet: Properties): void {
-    this.properties = {
-      ...Object.keys(propertiesToSet).map((propertyName) => [
-        propertyName,
-        get(propertyName),
-      ]),
-      ...this.properties,
-    };
     Object.entries(propertiesToSet).forEach(([propertyName, propertyValue]) => {
+      if (!((propertyName as KnownProperty) in this.properties)) {
```
Comment (phulin) [inline src/property.ts:219]: I prefer o[k] === undefined for this pattern, and I think that's what we use elsewhere.

**Review by phulin — CHANGES_REQUESTED** [review]

File: src/property.ts (hunk 2 above)
Comment (horrible-little-slime) [reply src/property.ts:227]: I originally did that, but my concern was that I wasn't sure the object I was constructing in the way I set it up would actually act as a Properties thing. I've changed it to use this.set, but make sure to give that section a careful look to make sure it's not just entirely wrong

File: src/property.ts [hunk 4]
```diff
@@ -215,23 +215,21 @@ export class PropertiesManager {
...
-        propertyName,
-        get(propertyName),
-      ]),
-      ...this.properties,
-    };
-    Object.entries(propertiesToSet).forEach(([propertyName, propertyValue]) => {
+    for (const [propertyName, propertyValue] of Object.entries(
+      propertiesToSet
+    )) {
+      if (this.properties[propertyName as KnownProperty] !== undefined) {
```
Comment (phulin) [inline src/property.ts:221]: shouldn't this only be if it DOES equal undefined? I.e. save a property's value if it's the first time we are setting it in the session?

**Review by phulin — CHANGES_REQUESTED** [review]

File: src/property.ts (hunk 4 above)
Comment (horrible-little-slime) [reply src/property.ts:221]: yep, force of habit. Whoops!

**Review by phulin — APPROVED** [review]

**Pull request merged**